dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.hdlresolver;
+
+import java.text.MessageFormat;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller is public and is useful for handle resolving,
+ * wheter a target handle identifier will be resolved into the
+ * corresponding URL (if found), otherwise will respond a null string.
+ * 
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca at 4science.it)
+ *
+ */
+@RestController
+@RequestMapping(path = "/{hdlService:hdlResolver|resolve|listhandles|listprefixes}/")
```

Comment:
Could you add a REST Contract for this new endpoint?  That way it's fully described.

I also feel this is necessary because the `@RequestMapping` looks odd here. Unless I'm misunderstanding the syntax, it appears this creates several top-level paths?  `/api/hdlResolver/`, `/api/resolve/`, etc.  Why are these not structured similar to other endpoints?  It seems like they all should be grouped under `/api/hdlResolver/_subpath_` or something similar?

Additionally, we don't usually use camel-case to define endpoints.  So, `hdlResolver` may need to become `hdlresolver` or `handleresolver`